Incorrect use of fsnotify in configtls

**Describe the bug**
When working on #8395, I observed another testing failure:

```
=== RUN   TestErrorRecordedIfFileDeleted
Event: fsnotify.Event{Name:"/var/folders/9l/f2cn0_n52w38y4m77wms60gc0000gn/T/clientCAs.crt1808838314", Op:0x12}    clientcasfilereloader_test.go:65:
        	Error Trace:	/Users/haoqixu/code/personal/wip/otel/opentelemetry-collector/config/configtls/clientcasfilereloader_test.go:65
        	Error:      	Condition never satisfied
        	Test:       	TestErrorRecordedIfFileDeleted
    clientcasfilereloader_test.go:70:
        	Error Trace:	/Users/haoqixu/code/personal/wip/otel/opentelemetry-collector/config/configtls/clientcasfilereloader_test.go:70
        	Error:      	Not equal:
        	            	expected: "test error on reload"
        	            	actual  : "<nil>"

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-test error on reload
        	            	+<nil>
        	Test:       	TestErrorRecordedIfFileDeleted
--- FAIL: TestErrorRecordedIfFileDeleted (5.00s)
```

**Steps to reproduce**
Run `go test -v -run TestErrorRecordedIfFileDeleted -race -count 1000 .` under `config/configtls`.

**Environment**
OS: macOS 13.2.1
Compiler(if manually compiled): go1.20.3

**Additional context**

The `Op` field of [`fsnotify.Event`](https://pkg.go.dev/github.com/fsnotify/fsnotify#Event) is a bitmask and some systems may send multiple operations at once. We should use the `Event.Has()` method instead of comparing with `==`.

The failure above was caused by an multiple operations event.

--- a/.chloggen/fix-incorrect-use-of-fsnotify.yaml
+++ b/.chloggen/fix-incorrect-use-of-fsnotify.yaml
@@ -1,0 +1,25 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: configtls
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: fix incorrect use of fsnotify
+
+# One or more tracking issues or pull requests related to the change
+issues: [8438]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []

--- a/config/configtls/clientcasfilereloader.go
+++ b/config/configtls/clientcasfilereloader.go
@@ -112,7 +112,7 @@ func (r *clientCAsFileReloader) handleWatcherEvents() {
 			// NOTE: k8s configmaps uses symlinks, we need this workaround.
 			// original configmap file is removed.
 			// SEE: https://martensson.io/go-fsnotify-and-kubernetes-configmaps/
-			if event.Op == fsnotify.Remove || event.Op == fsnotify.Chmod {
+			if event.Has(fsnotify.Remove) || event.Has(fsnotify.Chmod) {
 				// remove the watcher since the file is removed
 				if err := r.watcher.Remove(event.Name); err != nil {
 					r.lastReloadError = err
@@ -123,7 +123,7 @@ func (r *clientCAsFileReloader) handleWatcherEvents() {
 				}
 				r.reload()
 			}
-			if event.Op == fsnotify.Write {
+			if event.Has(fsnotify.Write) {
 				r.reload()
 			}
 		}

